students/thorn/lesson06/mailroom_6.py

```diff
@@ -0,0 +1,179 @@
+import os
+
+"""
+Thomas Horn
+mailroom.py
+Tracks donor information and automates the sending of thank you notes.
+"""
+donors = {
+    'Tom Horn': [599.23, 1000.00],
+    'Theo Hartwell': [0.01, 0.01, 0.1],
+    'Bailey Kimmitt': [8723.22, 27167.22, 91817.66],
+    'Paul Hubbell': [90012.32, 2312.24],
+    'David Beckham': [1817266.11, 123123.66, 111335.112]
+}
+
+def get_donor_names():
+    """ Returns the donor keys - representing the donor names. """
+    return donors.keys()
+
+
+def get_donor_totals(donor):
+    """ Returns the total donation for a donor.  Not currently active. Except for testing. """
+    print(sum(donors[donor]))
+    return(sum(donors[donor]))
+
+
+def print_thanks(name_request, donation_amount):
+    message = f"""Dear {name_request},
+        Thank you for you generous donation of ${donation_amount:.2f}.
+        It will truly help the children.
+
+        Sincerely,
+        Donation Recievers"""
+    print(message)
+
+
+def send_thanks():
+    """
+    Prompts user for a full name and performs an action based on the input.
+    - 'list' -> shows a list of donor names and re-prompts
+    - Name not in list -> adds the name to the data structure and uses it
+    - Name in list -> use it
+    After name is used -> prompt for donation amount and turn into a number.
+    - Add that amount to donation history of selected user
+    - Print thank you note
+    - Returns to original prompt or quit to prompt
+    """
+    # Exception block test -> see if they entered an actual titled name
+    # Ignore the fact that it lowercases the name later
+    try:
+        name_request = input("Please enter a donor's full name, or 'list' for a full list of donors.  'Quit' or 'Return' valid commands. ")
+        name_request[0] = name_request.title[1]
+    except:
+        print("Names start with an uppercase letter here.")
+        name_request = name_request.title()
+        
+    # Return or Quit.
+    if name_request.lower() == "quit":
+        quit()
+
+    if name_request.lower() == "return":
+        return
+
+    # List request - list of donors returned
+    # Exception added to make user format it properly.
+    if name_request.lower() == 'list':
+        # print('\n'.join(get_donor_names()))
+        donor_names = get_donor_names()
+        print('\n'.join(donor_names))
+        return
+
+    name_request = name_request.title()
+    donation_amount = float(input(("Please enter a donation amount for {}: ".format(name_request))))
```

Comment:
This would be a good place for a try/except block to catch a ValueError for non-numerical input